ledger-tool/src/blockstore.rs

```diff
@@ -709,6 +710,47 @@ fn do_blockstore_process_command(ledger_path: &Path, matches: &ArgMatches<'_>) -
             let starting_slot = value_t_or_exit!(arg_matches, "starting_slot", Slot);
             for slot in blockstore.duplicate_slots_iterator(starting_slot)? {
                 println!("{slot}");
+                if verbose_level > 0 {
```

Comment:
nit: Would you mind moving the logic to a function defined in `output.rs`? I'm thinking that with the sub-function, we'll look something like the `slot` command where just loop over slots and call `output_duplicate_slot_proof()`:
https://github.com/anza-xyz/agave/blob/ae021442a06e220c84abd70f73c39a50f582ddc4/ledger-tool/src/blockstore.rs#L1024-L1029

The general idea here is to keep the logic near `match.subcommand()` more minimal. There are some functions that need cleanup (which is obviously low priority), but better to make new functionality match where we want to go.

Reply:
If you want to get real fancy, you could define a `CliDuplicateSlotProof` type that implements `Display` and `From<DuplicateSlotProof>`, altho `From` may not work well with variable verbosity.

The pros of this is that the logic would be contained in the type, and then we'd also support `--output json` for free by deriving the Serde traits:

```rust
#[derive(Deserialize, Serialize, Display)
pub struct CliDuplicateSlotProof {
    shred1: CliDuplicateShred,
    shred2: CliDuplicateShred,
}

#[derive(Deserialize, Serialize)
pub struct CliDuplicateShred {
    fetc_set_index: ...,
    ...
}
...
```
Check out `SlotBounds` for an example where we did this once already:
https://github.com/anza-xyz/agave/blob/be16321eb0db3e12a57a32f59febbf54b92ebb7c/ledger-tool/src/output.rs#L50-L52

And note the satisfying one-liner to print, whether it is output as text or json:
https://github.com/anza-xyz/agave/blob/be16321eb0db3e12a57a32f59febbf54b92ebb7c/ledger-tool/src/blockstore.rs#L664

Reply:
Nice idea went the fancy route, and hijacked VerboseDisplay to get the variable verbosity